modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundKernelOpsEvent.java

```diff
@@ -127,7 +127,9 @@ public void startMessageDelivery() {
      * Stop message delivery threads in Andes
      */
     public void stopMessageDelivery() {
-        messagingEngine.stopMessageDelivery();
+        if(messagingEngine != null){
```

Comment:
Can messagingEngine be null at this point? if it is null why are we calling "stopMessageDelivery"?